docs/components/dropdown-menu.md

```diff
@@ -438,13 +481,43 @@ There are 3 primary item types:
 </KDropdownMenu>
 ```
 
+### Slots
+
+#### action
+
+Slot for any action triggers that you would want to keep separate from your items. Mouse events dispatched by contents of this slot will not be propagated to the item that wraps it, so you can handle item clicks and action clicks separately (see example above for reference).
+
+Clicks on this slot content also don't trigger dropdown close, so you can handle it yourself using `closeDropdown` slop prop exposed by [`items`](#items-1) slot.
+
+```html
+<KDropdownItem
+  @click="clickHandler"
+  @action:click="closeDropdown"
+>
+  Button w/ action
+  <template #action>
+    <KButton
+      appearance="btn-link"
+      @click="actionClickHandler"
+    >
+      <KIcon icon="trash" />
+    </KButton>
+  </template>
+</KDropdownItem>
+```
+
+:::warning NOTE
+This slot is only provided by `KDropdownItem` component, not the `KDropdownMenu` itself.
+:::
+
 ### Events
 
-| Event     | Description         |
-| :-------- | :------------------ |
-| `@click` | Fires when a `button` type menu item is clicked |
-| `@change` | Fires when items within a `selectionMenu` are clicked; returns the selected menu item object or `null` |
-| `@toggleDropdown` | Fires when the button to toggle the menu is clicked; returns `true` if the menu is open, or `false` |
+| Event             | Description                                                                                                              |
+| :---------------- | :----------------------------------------------------------------------------------------------------------------------- |
+| `@click`          | Fires when a `button` type menu item is clicked                                                                          |
+| `@change`         | Fires when items within a `selectionMenu` are clicked; returns the selected menu item object or `null`                   |
+| `@toggleDropdown` | Fires when the button to toggle the menu is clicked; returns `true` if the menu is open, or `false`                      |
+| `@action:click`   | Emitted only by by the `KDropdownItem` component. Fires when content passed through [`action`](#action) slot is clicked. |
```

Comment:
question: why emit a namespaced event when there's nothing else going on? Can't this just be `@click` or `@dropdown-item-click`?

Reply:
Because `@click` is already emitted by `KDropdownItem`. I think `action:click` is the most intuitive for click on `action` slot **specifically**.
```
<KDropdownItem
  @click="clickHandler"
  @action:click="closeDropdown"
>...
```